src/routes/index.tsx

```diff
@@ -1,15 +1,20 @@
 import React from 'react';
 import { useRoutes } from 'react-router-dom';
 
-import { Landing } from '@features/landing';
-import { Login } from '@features/login';
-import { Register } from '@features/register';
+import { Landing } from '@features/Landing';
+import { useSession } from '@providers/SessionContext';
+
+import { protectedRoutes } from './protected';
+import { publicRoutes } from './public';
 
 export const AppRoutes = () => {
-  const routes = [
-    { path: '/', element: <Landing /> },
-    { path: '/login', element: <Login /> },
-    { path: '/sign-up', element: <Register /> },
-  ];
-  return useRoutes(routes);
+  const { isAuthenticated } = useSession();
+
+  const commonRoutes = [{ index: true, element: <Landing /> }];
+
+  const routes = isAuthenticated ? protectedRoutes : publicRoutes;
+
+  const element = useRoutes([...routes, ...commonRoutes]);
+
+  return <>{element}</>;
```

Comment:
I think it looks cleaner but just a suggestion:
```suggestion
  const { isAuthenticated } = useSession();

  const commonRoutes = [{ index: true, element: <Landing /> }];
  const routes = isAuthenticated ? protectedRoutes : publicRoutes;
  const element = useRoutes([...routes, ...commonRoutes]);

  return <>{element}</>;
```